### ISSUE TITLE
Is there any maintained fork

### ISSUE THREAD

**Issue by iamareebjamal** [miLibris/flask-rest-jsonapi#188]
I realize the tooling for JSONAPI is not very popular, but we have to maintain our code which is using the library heavily. It's known that this repo is not maintained anymore and there are a lot of issues in it. Does anyone have a link to a maintained fork? Or is there any maintained library with migration path?

If not, then community can decide to fork the project and volunteers can help in getting it up to date?

**Comment by TMiguelT** [comment]
Hi, I've been maintaining a fork over at https://github.com/TMiguelT/flapison. I've made and merged quite a number of PRs. Please feel free to contribute.

Unfortunately it's notoriously difficult to publicise a fork. I don't even think Google indexes them. So if you can think of a good way for me to do that, let me know.

**Comment by iamareebjamal** [comment]
Can you request it to be added here? https://jsonapi.org/implementations/#server-libraries-python

Seeing that JSONAPI is not that popular, this is the most useful resource to look at when searching for server implementations

**Comment by TMiguelT** [comment]
Oh, that's a good idea. I'll see if I can PR that into the jsonapi docs.

**Comment by TMiguelT** [comment]
PR submitted. Hopefully it will get merged and it will clear things up.

**Comment by akira-dev** [comment]
Hello, I will have some time to review PR during next week I hope. So new release is coming soon. I am very sorry for my lack of time to maintain this repo.

**Comment by TMiguelT** [comment]
Oh, hi! While you're around, is there any way I can help with the maintenance of this library? I'm happy to help review and merge PRs etc, or alternatively if you have time now would you accept a PR containing all the downstream changes from flapison? It contains all the PRs from me into this library (7 of them), and some other changes like running black on the whole repo. I understand if you don't want all of my changes.

**Comment by iamareebjamal** [comment]
Hi, it would also make sense to pick some contributors (@TMiguelT) is extremely active to maintain the library when you are inactive. This will concentrate the efforts in one place and derisk the library as it will be a community effort then?

**Comment by mahenzon** [comment]
Hi there! @TMiguelT I'm glad to see you here (I think we've met somewhere within marshmallow-jsonapi).
My colleague (@Znbiz) and I have been working on a fork too.

The fact is that we've developed it for our production. Our goal here was to achieve full compatibility with our huge legacy codebase, so we can rewrite the existing API by one resource at a time. At that point we faced the fact, that we need:
- JSON:API spec (a standard, finally!)
- api spec swagger documentation (that helps a lot both frontend and backend developers!)
- some convenient serialization / deserialization (marshmallow fits here pretty good)
- custom sorting and filtering, that can be defined outside of the lib
- an advanced tool to manage permissions for certain fields, not only routes
- a tool for creating RPC resources (when the backend is already designed this way, there's no way to somehow rewrite it to be compatible with the new modern and shiny spec, so we need to create custom non-jsonapi resources as well as regular resources)
- a centralized tool for working with PostgreSQL's JSONB fields

At this point we could say, that it needs a lot of work.

So, we came up to this solution:
- Fork `flask-rest-jsonapi` and create minimal changes, just to make it compatible with the **plugin system** and create some other whistles like advanced sorting and filtering
- Create a separate library for all our requirements, that can exist out the main library

### Finally!
There's https://github.com/AdCombo/flask-rest-jsonapi that has these improvenets:

- support for plugins
- support for custom filtering https://github.com/AdCombo/flask-rest-jsonapi/blob/725cf11286a1ea06aa2ca62182a212d03fceb1c8/flask_rest_jsonapi/data_layers/filtering/alchemy.py#L48-L81
- and also for custom sorting https://github.com/AdCombo/flask-rest-jsonapi/blob/725cf11286a1ea06aa2ca62182a212d03fceb1c8/flask_rest_jsonapi/data_layers/sorting/alchemy.py#L48-L72

(for custom filtering/sorting you need to define a new method in your marshmallow field as described in docstrings)

- some other bugfixes


And a separate library https://github.com/AdCombo/combojsonapi/, which has:
- [ApiSpec Plugin](https://github.com/AdCombo/combojsonapi/blob/master/docs/en/api_spec_plugin.rst) - that's swagger documentation based on marshmallow's apispec
- [Permission Plugin](https://github.com/AdCombo/combojsonapi/blob/master/docs/en/permission_plugin.rst) for managing access to certain fields
- [Event Plugin](https://github.com/AdCombo/combojsonapi/blob/master/docs/en/event_plugin.rst) - for creating custom RPC resources
- [PostgreSqlJSONB Plugin](https://github.com/AdCombo/combojsonapi/blob/master/docs/en/postgresql_jsonb.rst) for easing out work with PostgreSQL's JSONB fields from frontend


___


As I see, there's already some work going in the https://github.com/TMiguelT/flapison, that's cool. So I think maybe we could discuss here some suggestions / ideas how we all together can produce a better library, based on the `flask-rest-jsonapi`, that fit's all developers' needs, is actively maintained and has cool and useful features!

### Looking forward to hearing from you!

**Comment by TMiguelT** [comment]
Hi @mahenzon. It looks like you've done some excellent work refactoring flask-rest-jsonapi. Plugins are a great idea, and moving a lot of the SQLAlchemy data layer into a shared module is something I was meaning to do myself.

We should obviously try to merge the two forks. Your changes are much more fundamental so I'm happy to try to rebase my changes against yours and make some PRs. 

I have one reservation, however. I'm a tad sceptical of open source projects maintained under company organisations, because they tend to languish once the initial development effort is finished, and stop paying attention to issues and PRs. After all, that's what's happened with [miLibris](https://github.com/miLibris) who originally wrote this. Do you have any thoughts on how maintenance might continue in the long term? Would you be open to adding collaborators even to your organisation repo? If not I can see similar problems to `miLibris/flask-rest-jsonapi` happening in the future.

**Comment by mahenzon** [comment]
Hi again @TMiguelT!

I'm really glad to see that you like it and are interested in collaborating!
And about your concerns regarding maintenance: we welcome collaborators to our repos! Sure, no one is interested in another stalled fork.

I've just invited you to both https://github.com/AdCombo/flask-rest-jsonapi and https://github.com/AdCombo/combojsonapi as a collaborator with write access. Feel free to push changes and open PRs. We definitely want to see some improvements and welcome collaborators! It'll be cool to see your changes from flapison

**Comment by TMiguelT** [comment]
Okay that sounds great. I'll try to find time to put in some PRs before too long. Thanks for the collaboration invitations also, I can't accept them just yet because of 2FA issues but I should be able to eventually.

One other thought, do you think it would be helpful to rename your repo from `flask-rest-jsonapi` so that it's easier to search and/or allows it to be added to the JSON:API site as a separate project? Those are the main reasons I renamed my fork. I also think it makes it clearer to users that it's not the exact same project, especially in case your API diverges from upstream.

**Comment by mahenzon** [comment]
Yeah, I agree. It's definitely doable. We just need to come up with some new name. Sure, it'll help in different ways. Give us a day to deside. The results will be in this issue https://github.com/AdCombo/flask-rest-jsonapi/issues/10

Waiting for you to contribute, thanks!

**Comment by buckmaxwell** [comment]
It seems perhaps this repository is being maintained? what is the latest on this.

**Comment by mahenzon** [comment]
hi @buckmaxwell!
this fork is actively maintained https://github.com/AdCombo/flask-combo-jsonapi

more detailed in the [comment above](https://github.com/miLibris/flask-rest-jsonapi/issues/188#issuecomment-616794803)
